File: fix_imports.py
import os
import re

root_dir = r"c:\Projects\KUMC\Figma_NextJs_Code\KUMC-PPP-Dev-Version\src"
exclude_dirs = {"node_modules", ".next"}

# Regex to find @version before end of string in import
# Matches @ followed by digits/dots if they are followed by ' or "
pattern = re.compile(r"(@\d+(\.\d+)*)(?=['\"])")

def fix_imports():
    for root, dirs, files in os.walk(root_dir):
        # Exclude specific directories
        dirs[:] = [d for d in dirs if d not in exclude_dirs]
        
        for file in files:
            if file.endswith(".tsx") or file.endswith(".ts"):
                file_path = os.path.join(root, file)
                try:
                    with open(file_path, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    new_content = pattern.sub('', content)
                    
                    if new_content != content:
                        with open(file_path, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Fixed imports in: {file_path}")
                except Exception as e:
                    print(f"Error processing {file_path}: {e}")

if __name__ == "__main__":
    fix_imports()

File: fix_imports.js
const fs = require('fs');
const path = require('path');

const rootDir = 'c:\\Projects\\KUMC\\Figma_NextJs_Code\\KUMC-PPP-Dev-Version\\src';
const excludeDirs = new Set(['node_modules', '.next']);

const pattern = /(@\d+(\.\d+)*)(?=['"])/g;

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (!excludeDirs.has(file)) {
                walk(fullPath);
            }
        } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
            try {
                const content = fs.readFileSync(fullPath, 'utf8');
                const newContent = content.replace(pattern, '');
                if (newContent !== content) {
                    fs.writeFileSync(fullPath, newContent, 'utf8');
                    console.log(`Fixed imports in: ${fullPath}`);
                }
            } catch (err) {
                console.error(`Error processing ${fullPath}: ${err.message}`);
            }
        }
    }
}

walk(rootDir);
